libs/api_common/windows_platform_common.cpp

```diff
@@ -716,41 +716,39 @@ _load_all_program_data_information()
 static void
 _load_ebpf_provider_data()
 {
-    std::unique_lock lock(_windows_program_information_lock);
-
-    if (!_windows_program_information.empty()) {
-        return;
-    }
-
-    ebpf_result_t result = _load_all_program_data_information();
-    if (result != EBPF_SUCCESS) {
-        goto Exit;
-    }
-
-    result = _load_all_section_data_information();
-    if (result != EBPF_SUCCESS) {
-        goto Exit;
-    }
+    try {
+        std::call_once(*_windows_program_information_init_flag, [] {
```

Comment:
question: does `std:call_once` do an initial lock-free check?

Reply:
https://en.cppreference.com/w/cpp/thread/call_once 

![image](https://github.com/microsoft/ebpf-for-windows/assets/20480683/624cd91f-d9ea-4dc0-b710-7e6078c3e30a)